File: src/main/java/com/jozufozu/exnihiloomnia/common/util/MathStuff.java
package com.jozufozu.exnihiloomnia.common.util;

public class MathStuff
{
    public static float lerp(float f1, float f2, float fraction)
    {
        return (1.0f - fraction) * f1 + f2 * fraction;
    }
    
    public static double lerp(double f1, double f2, double fraction)
    {
        return (1.0 - fraction) * f1 + f2 * fraction;
    }
}

File: src/main/kotlin/com/jozufozu/exnihiloomnia/common/util/MathStuff.kt
package com.jozufozu.exnihiloomnia.common.util

object MathStuff {
    fun lerp(f1: Float, f2: Float, fraction: Float): Float {
        return (1.0f - fraction) * f1 + f2 * fraction
    }

    fun lerp(f1: Double, f2: Double, fraction: Double): Double {
        return (1.0 - fraction) * f1 + f2 * fraction
    }
}
